cplusplus/UnconvertFromZDW.cpp

```diff
@@ -926,12 +1015,36 @@ ERR_CODE UnconvertFromZDW_Base::readHeader()
 	if (this->version > UNCONVERT_ZDW_VERSION)
 		return UNSUPPORTED_ZDW_VERSION_ERR;
 
-	if (this->version == 1)
-		this->decimalFactor = DECIMAL_FACTOR_VERSION_1;
+	if (this->bShowBasicStatisticsOnly)
+		this->statusOutput(INFO, "File version %d\n", static_cast<int>(this->version));
+
+	this->metadata.clear();
+	if (this->version >= 11)
+	{
+		//2. Metadata block
+		ULONG metadata_size;
+		readBytes(&metadata_size, 4);
+
+		if (this->bShowBasicStatisticsOnly)
+			this->statusOutput(INFO, "Metadata block size = %u\n", metadata_size);
+
+		char *metadata_block = static_cast<char*>(malloc(metadata_size));
+		readBytes(metadata_block, metadata_size);
+		const char *key = metadata_block, *value;
+		while (key - metadata_block < static_cast<long>(metadata_size)) {
+			value = key + strlen(key) + 1; //skip past null terminator
+			this->metadata[string(key)] = string(value);
 
-	//2. Parse file attributes here (before version 3).
-	if (this->version <= 2)
+			//advance to next key-value pair
+			key = value + strlen(value) + 1; //skip past null terminator
+		}
+		free(metadata_block);
+	} else if (this->version <= 2)
 	{
```

Comment:
nit: no other place has linebreaks around braces like this (i.e. "} ...\n{".  perhaps pull out the newline?  (though I admit, this is less bothersome to me... at least while out whitespace around control-flow braces is so schizophrenic)

Reply:
Adjusted to align with other uses of "else if" in the code.